stablehlo/tests/ops_speculatability.mlir

```diff
@@ -485,42 +485,41 @@ func.func @slice(%static_arg: tensor<2xf64>, %dynamic_arg: tensor<?xf64>) {
 // CHECK-LABEL: func @all_reduce
 // CHECK-NEXT:  return
 func.func @all_reduce(%static_arg: tensor<2xf64>, %dynamic_arg: tensor<?xf64>) {
-  %speculatable_0 = "stablehlo.all_reduce"(%static_arg) ({
+  %0 = "stablehlo.all_reduce"(%static_arg) ({
```

Comment:
I actually prefer this format to test specularity after each op, but since the majority of the related tests list all the ops first and then the specularity stuff, we can keep this format too, but let's use one format consistently across the tests.

Reply:
SG, let me just change everything to test after each op then. I like having the various test cases grouped because it makes it easier to see the differences. But with ops that take more than a few lines, it becomes hard to see the speculatability checks. Since the speculatability are just one line though, it doesn't hurt visual diffability too much to always put the checks right after the ops they are checking.

Done.